### ISSUE TITLE
Renewal never works, now getting "Requirement.parse('certbot==0.9.0.dev0'))" error

### ISSUE THREAD

**Issue by AKAMEDIASYSTEM** [certbot/certbot#5261]
## My operating system is (include version):
Ubuntu 14.04

## I installed Certbot with (certbot-auto, OS package manager, pip, etc):
sudo apt-get install python-certbot-nginx (per https://www.nginx.com/blog/using-free-ssltls-certificates-from-lets-encrypt-with-nginx/)

## I ran this command and it produced this output:
sudo certbot --nginx -d aka.farm -d craft.aka.farm -d weft.aka.farm

## Certbot's behavior differed from what I expected because:
`An unexpected error occurred:
VersionConflict: (certbot 0.19.0 (/usr/lib/python2.7/dist-packages), Requirement.parse('certbot==0.9.0.dev0'))
Please see the logfile '/tmp/tmpZH1KEb' for more details.`

###### Logs are stored in `/var/log/letsencrypt` by default. Feel free to redact domains, e-mail and IP addresses as you see fit.
`
# cat /var/log/letsencrypt/letsencrypt.log
2017-11-24 12:07:02,945:DEBUG:certbot.main:certbot version: 0.19.0
2017-11-24 12:07:02,945:DEBUG:certbot.main:Arguments: ['--quiet']
2017-11-24 12:07:02,945:DEBUG:certbot.main:Discovered plugins: PluginsRegistry(PluginEntryPoint#apache,PluginEntryPoint#manual,PluginEntryPoint#nginx,PluginEntryPoint#null,PluginEntryPoint#standalone,PluginEntryPoint#webroot)
2017-11-24 12:07:02,983:DEBUG:certbot.log:Root logging level set at 30
2017-11-24 12:07:02,984:INFO:certbot.log:Saving debug log to /var/log/letsencrypt/letsencrypt.log
2017-11-24 12:07:03,039:DEBUG:certbot.plugins.selection:Requested authenticator <certbot.cli._Default object at 0x7f58a2593cd0> and installer <certbot.cli._Default object at 0x7f58a2593cd0>
2017-11-24 12:07:03,039:DEBUG:certbot.cli:Default Detector is Namespace(account=<certbot.cli._Default object at 0x7f58a258d910>, agree_dev_preview=None, allow_subset_of_names=<certbot.cli._Default object at 0x7f58a258d6d0>, apache=<certbot.cli._Default object at 0x7f58a2593750>, apache_challenge_location=<certbot.cli._Default object at 0x7f58a25dba90>, apache_ctl=<certbot.cli._Default object at 0x7f58a2599c10>, apache_dismod=<certbot.cli._Default object at 0x7f58a258d550>, apache_enmod=<certbot.cli._Default object at 0x7f58a258d790>, apache_handle_modules=<certbot.cli._Default object at 0x7f58a25db290>, apache_handle_sites=<certbot.cli._Default object at 0x7f58a25c4f50>, apache_init_script=<certbot.cli._Default object at 0x7f58a25998d0>, apache_le_vhost_ext=<certbot.cli._Default object at 0x7f58a258d3d0>, apache_logs_root=<certbot.cli._Default object at 0x7f58a25dbc50>, apache_server_root=<certbot.cli._Default object at 0x7f58a258d090>, apache_vhost_root=<certbot.cli._Default object at 0x7f58a25dbe50>, authenticator=<certbot.cli._Default object at 0x7f58a2593cd0>, break_my_certs=<certbot.cli._Default object at 0x7f58a258e750>, cert_path=<certbot.cli._Default object at 0x7f58a2599650>, certname=<certbot.cli._Default object at 0x7f58a25dbad0>, chain_path=<certbot.cli._Default object at 0x7f58a2599950>, checkpoints=<certbot.cli._Default object at 0x7f58a2599150>, config_dir=<certbot.cli._Default object at 0x7f58a2599a50>, config_file=None, configurator=<certbot.cli._Default object at 0x7f58a2593cd0>, csr=<certbot.cli._Default object at 0x7f58a2593f10>, debug=<certbot.cli._Default object at 0x7f58a258e050>, debug_challenges=<certbot.cli._Default object at 0x7f58a258e150>, deploy_hook=<certbot.cli._Default object at 0x7f58a2593790>, dialog=None, directory_hooks=<certbot.cli._Default object at 0x7f58a2593990>, dns_cloudflare=<certbot.cli._Default object at 0x7f58a258ed10>, dns_cloudxns=<certbot.cli._Default object at 0x7f58a258eb10>, dns_digitalocean=<certbot.cli._Default object at 0x7f58a258e910>, dns_dnsimple=<certbot.cli._Default object at 0x7f58a258e710>, dns_dnsmadeeasy=<certbot.cli._Default object at 0x7f58a258e510>, dns_google=<certbot.cli._Default object at 0x7f58a258e310>, dns_luadns=<certbot.cli._Default object at 0x7f58a258e110>, dns_nsone=<certbot.cli._Default object at 0x7f58a258ded0>, dns_rfc2136=<certbot.cli._Default object at 0x7f58a258dcd0>, dns_route53=<certbot.cli._Default object at 0x7f58a258dad0>, domains=<certbot.cli._Default object at 0x7f58a25db950>, dry_run=<certbot.cli._Default object at 0x7f58a25dbbd0>, duplicate=<certbot.cli._Default object at 0x7f58a258da10>, eff_email=<certbot.cli._Default object at 0x7f58a25dbfd0>, email=<certbot.cli._Default object at 0x7f58a25dbed0>, expand=<certbot.cli._Default object at 0x7f58a258d310>, force_interactive=<certbot.cli._Default object at 0x7f58a25db110>, fullchain_path=<certbot.cli._Default object at 0x7f58a2599850>, func=<function renew at 0x7f58a28201b8>, hsts=<certbot.cli._Default object at 0x7f58a258ec50>, http01_address=<certbot.cli._Default object at 0x7f58a258e650>, http01_port=<certbot.cli._Default object at 0x7f58a258e550>, ifaces=<certbot.cli._Default object at 0x7f58a2599450>, init=<certbot.cli._Default object at 0x7f58a2599250>, installer=<certbot.cli._Default object at 0x7f58a2593cd0>, key_path=<certbot.cli._Default object at 0x7f58a2599750>, logs_dir=<certbot.cli._Default object at 0x7f58a2599c50>, manual=<certbot.cli._Default object at 0x7f58a2593150>, manual_auth_hook=<certbot.cli._Default object at 0x7f58a258d8d0>, manual_cleanup_hook=<certbot.cli._Default object at 0x7f58a2599510>, manual_public_ip_logging_ok=<certbot.cli._Default object at 0x7f58a2599310>, max_log_backups=<certbot.cli._Default object at 0x7f58a25db450>, must_staple=<certbot.cli._Default object at 0x7f58a258e950>, nginx=<certbot.cli._Default object at 0x7f58a2593550>, nginx_ctl=<certbot.cli._Default object at 0x7f58a2599d90>, nginx_server_root=<certbot.cli._Default object at 0x7f58a25996d0>, no_bootstrap=<certbot.cli._Default object at 0x7f58a258dd10>, no_self_upgrade=<certbot.cli._Default object at 0x7f58a258dc10>, no_verify_ssl=<certbot.cli._Default object at 0x7f58a258e250>, noninteractive_mode=<certbot.cli._Default object at 0x7f58a25db2d0>, num=<certbot.cli._Default object at 0x7f58a2593c10>, os_packages_only=<certbot.cli._Default object at 0x7f58a258db10>, post_hook=<certbot.cli._Default object at 0x7f58a2593590>, pre_hook=<certbot.cli._Default object at 0x7f58a2593490>, pref_challs=<certbot.cli._Default object at 0x7f58a2593390>, prepare=<certbot.cli._Default object at 0x7f58a2599350>, quiet=True, reason=<certbot.cli._Default object at 0x7f58a2599050>, redirect=<certbot.cli._Default object at 0x7f58a258ea50>, register_unsafely_without_email=<certbot.cli._Default object at 0x7f58a25dbcd0>, reinstall=<certbot.cli._Default object at 0x7f58a258d210>, renew_by_default=<certbot.cli._Default object at 0x7f58a258d4d0>, renew_hook=<certbot.cli._Default object at 0x7f58a2593690>, renew_with_new_domains=<certbot.cli._Default object at 0x7f58a258d5d0>, rsa_key_size=<certbot.cli._Default object at 0x7f58a258e850>, server=<certbot.cli._Default object at 0x7f58a2593ed0>, staging=<certbot.cli._Default object at 0x7f58a258df10>, standalone=<certbot.cli._Default object at 0x7f58a2593350>, standalone_supported_challenges=<certbot.cli._Default object at 0x7f58a2599e90>, staple=<certbot.cli._Default object at 0x7f58a2593090>, strict_permissions=<certbot.cli._Default object at 0x7f58a2593290>, text_mode=<certbot.cli._Default object at 0x7f58a25db5d0>, tls_sni_01_address=<certbot.cli._Default object at 0x7f58a258e450>, tls_sni_01_port=<certbot.cli._Default object at 0x7f58a258e350>, tos=<certbot.cli._Default object at 0x7f58a258d810>, uir=<certbot.cli._Default object at 0x7f58a258ee50>, update_registration=<certbot.cli._Default object at 0x7f58a25dbdd0>, user_agent=<certbot.cli._Default object at 0x7f58a2593d10>, user_agent_comment=<certbot.cli._Default object at 0x7f58a2593e10>, validate_hooks=<certbot.cli._Default object at 0x7f58a2593890>, verb='renew', verbose_count=<certbot.cli._Default object at 0x7f58a25dba50>, webroot=<certbot.cli._Default object at 0x7f58a258ef10>, webroot_map=<certbot.cli._Default object at 0x7f58a259c0d0>, webroot_path=<certbot.cli._Default object at 0x7f58a2599c90>, work_dir=<certbot.cli._Default object at 0x7f58a2599b50>)
2017-11-24 12:07:03,054:INFO:certbot.renewal:Cert not yet due for renewal
2017-11-24 12:07:03,054:DEBUG:certbot.renewal:no renewal failures
`

## Here is the relevant nginx server block or Apache virtualhost for the domain I am configuring:
I followed this setup guide initially: https://www.digitalocean.com/community/tutorials/how-to-secure-nginx-with-let-s-encrypt-on-ubuntu-14-04
...but renewal has never worked.

I'm opening this issue because I don't understand how to resolve the issue that certbot apparently is checking to see if it has a very outdated (0.9.0) version, which is not available in the ppa:certbot/certbot repository. Then again, certbot fails at every attempt to auto-renew, so something larger may be at issue.
Thanks in advance for any assistance you can provide.

**Comment by SwartzCr** [comment]
I think this likely means that you've installed some part of certbot not from the PPA - can you share the output of `pip freeze`?
Also of `dpkg -l | grep certbot`?

**Comment by AKAMEDIASYSTEM** [comment]
Thank you for the reply! I think you are right about my having obtained certbot from somewhere else.
`pip freeze` shows this:
`
certbot==0.19.0
certbot-nginx==0.9.0.dev0
`
and `dpkg -l | grep certbot` shows:
`
ii  certbot                             0.19.0-1+ubuntu14.04.1+certbot+1           all          automatically configure HTTPS using Let's Encrypt
rc  letsencrypt                         0.17.0-2+ubuntu14.04.1+certbot+1           all          transitional dummy package
ii  python-acme                         0.19.0-1+ubuntu14.04.1+certbot+1           all          ACME protocol library for Python 2
ii  python-certbot                      0.19.0-1+ubuntu14.04.1+certbot+1           all          main library for certbot
ii  python-certbot-nginx                0.19.0-1+ubuntu14.04.1+certbot+1           all          Nginx plugin for Certbot
ii  python-cffi-backend                 1.10.0-0+certbot~trusty+1                  amd64        Foreign Function Interface for Python calling C code - backend
ii  python-configargparse               0.11.0-1+certbot~trusty+1                  all          replacement for argparse with config files and environment variables (Python 2)
ii  python-cryptography                 1.7.1-2+certbot~trusty+1                   amd64        Python library exposing cryptographic recipes and primitives (Python 2)
ii  python-dialog                       3.3.0-2+certbot~trusty+1                   amd64        Python 2 module for making simple terminal-based user interfaces
ii  python-dnspython                    1.15.0-1+certbot~trusty+1                  all          DNS toolkit for Python
ii  python-idna                         2.2-1+certbot~trusty+1                     all          Python IDNA2008 (RFC 5891) handling (Python 2)
ii  python-ipaddress                    1.0.17-1+certbot~trusty+1                  all          Backport of Python 3 ipaddress module (Python 2)
ii  python-ndg-httpsclient              0.4.2-1+certbot~trusty+1                   all          enhanced HTTPS support for httplib and urllib2 using PyOpenSSL for Python2
ii  python-openssl                      17.0.0-0+certbot~trusty+1                  all          Python 2 wrapper around the OpenSSL library
ii  python-parsedatetime                2.1-3+certbot~trusty+1                     all          Python module to parse human-readable date/time expressions
ii  python-pkg-resources                33.1.1-1+certbot~trusty+1                  all          Package Discovery and Resource Access using pkg_resources
ii  python-psutil                       5.0.1-1+certbot~trusty+2                   amd64        module providing convenience functions for managing processes
ii  python-pyasn1                       0.1.9-2+certbot~trusty+1                   all          ASN.1 library for Python (Python 2 module)
ii  python-requests                     2.12.4-1+certbot~trusty+1                  all          elegant and simple HTTP library for Python2, built for human beings
ii  python-rfc3339                      1.0-4+certbot~trusty+1                     all          parser and generator of RFC 3339-compliant timestamps (Python 2)
ii  python-setuptools                   33.1.1-1+certbot~trusty+1                  all          Python Distutils Enhancements
ii  python-six                          1.10.0-3+certbot~trusty+1                  all          Python 2 and 3 compatibility library (Python 2 interface)
ii  python-urllib3                      1.19.1-1+certbot~trusty+1                  all          HTTP library with thread-safe connection pooling for Python
ii  python3-certifi                     2017.4.17-2+ubuntu14.04.1+certbot+1        all          root certificates for validating SSL certs and verifying TLS hosts (python3)
ii  python3-chardet                     3.0.4-1+ubuntu14.04.1+certbot+2            all          universal character encoding detector for Python3
ii  python3-idna                        2.5-1+ubuntu14.04.1+certbot+1              all          Python IDNA2008 (RFC 5891) handling (Python 3)
ii  python3-pkg-resources               33.1.1-1+certbot~trusty+1                  all          Package Discovery and Resource Access using pkg_resources
ii  python3-requests                    2.18.1-1+ubuntu14.04.1+certbot+1           all          elegant and simple HTTP library for Python3, built for human beings
ii  python3-setuptools                  33.1.1-1+certbot~trusty+1                  all          Python3 Distutils Enhancements
ii  python3-six                         1.11.0-1+ubuntu14.04.1+certbot+1           all          Python 2 and 3 compatibility library (Python 3 interface)
ii  python3-urllib3                     1.21.1-1+ubuntu14.04.1+certbot+1           all          HTTP library with thread-safe connection pooling for Python3
`
...embarrassingly, my certificates appear to have renewed without further effort on my part in the last day or so. It's great to have them up to date, but I don't feel great not understanding what went on :-(

Should I `sudo pip uninstall certbot-nginx` ?

Thanks again for the assistance! I'd be sunk without certbot...

AKA

**Comment by SwartzCr** [comment]
Removing certbot from pip might solve your issue if it's the case that your pip installed version is being called instead of your system packaged one. Removing the 0.9.0 version of certbot-nginx would be my first step. But you might also want to remove the 0.19.0 pip installed version in favor of the PPA version.
You might also want to remove the pip installed dependencies. pip uninstall doesn't do this by defauld, but pip-autoremove should: https://github.com/invl/pip-autoremove
Let me know what works for you, and hopefully you'll be error free soon.

**Comment by AKAMEDIASYSTEM** [comment]
Thanks again! 

I've removed certbot (both certbot-nginx and 0.19.0) and installed the PPA verison. I ran the `renew --dry-run` test and it appeared to work ok.

Thank you for helping me figure out what was going on with my different installs of certbot - I'm very grateful for the project, and especially for the noticeable improvement in documentation and support over the last year!

I think this ticket can be closed, since the issue was with me all along...

AKA
